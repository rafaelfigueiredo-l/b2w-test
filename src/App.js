import React, { Component } from "react";
import { createGlobalStyle } from "styled-components";
import AppAlert from "./components/app/AppAlert";
import AppHeader from "./components/app/AppHeader";
import Card from "./components/card/Card";
import background from "./assets/images/background.jpg";

class App extends Component {
  state = {
    alertError: {
      status: null,
      type: null,
      message: null,
      component: null,
    },
  };
  

  render() {
    const {alertError} = this.state;
    const {setAlert} = this;
    
    return (
      <>
        <GlobalStyle />
        <AppHeader/>
        <AppAlert alertError={alertError}/>
        <main id="AppContent">
          <Card setAlert={setAlert} />
        </main>
      </>
    );
  }

  setAlert = (type, message, component) => {
    this.setState({alertError: {
      status: true,
      type: type,
      message: message,
      component: component,
    }});
  }
}

const GlobalStyle = createGlobalStyle`
  * {
    -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
            box-sizing: border-box;
  }
  html, body, #app{
    padding: 0;
    margin: 0;
    min-height:100%;
    height:100%;
  }
  body {
    background:#000;
    background-image: url(${background});
    background-position: center;
    background-size:contain;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  #app{
    width:100%;
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
`;


export default App;
