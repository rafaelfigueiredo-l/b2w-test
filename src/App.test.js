import App from './App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot testing', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Component should have this default values', () => {
    const wrapper = shallow(<App />);
    let defaultValues = {
      status: null,
      type: null,
      message: null,
      component: null,
    };
    expect(wrapper.state('alertError')).toEqual(defaultValues);
  });

  it('alertError.status should be true', () => {
    const wrapper = shallow(<App />);
    const instance = wrapper.instance();
    instance.setAlert('Some Type', 'Some Message', 'Some Component');
    expect(instance.state.alertError).toEqual({
      status: true,
      type: 'Some Type',
      message: 'Some Message',
      component: 'Some Component',
    });
  });

  it('Should find AppAlert', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('AppAlert')).toHaveLength(1);
  });

  it('Should find AppHeader', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('AppHeader')).toHaveLength(1);
  });

  it('Should find Card', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('Card')).toHaveLength(1);
  });

});