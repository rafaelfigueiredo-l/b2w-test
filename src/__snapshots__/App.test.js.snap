// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshot testing 1`] = `
<Fragment>
  <Memo(GlobalStyleComponent) />
  <AppHeader />
  <AppAlert
    alertError={
      Object {
        "component": null,
        "message": null,
        "status": null,
        "type": null,
      }
    }
  />
  <main
    id="AppContent"
  >
    <Card
      setAlert={[Function]}
    />
  </main>
</Fragment>
`;
